src/store/conversations/conversations.epics.ts

```diff
@@ -1578,34 +1595,47 @@ const publishConversationEpic: AppEpic = (action$, state$) =>
     map(({ payload }) => ({
       publishRequest: payload,
       conversations: ConversationsSelectors.selectConversations(state$.value),
-      publishedFolders: ConversationsSelectors.selectSectionFolders(
-        state$.value,
-        PublishedWithMeFilter,
-      ),
+      publishedAndTemporaryFolders:
+        ConversationsSelectors.selectTemporaryAndFilteredFolders(
+          state$.value,
+          PublishedWithMeFilter,
+        ),
     })),
-    switchMap(({ publishRequest, conversations, publishedFolders }) => {
-      const sharedConversations = conversations
-        .filter((conversation) => conversation.id === publishRequest.id)
-        .map(({ folderId: _, ...conversation }) => ({
-          ...conversation,
-          ...resetShareEntity,
-          id: uuidv4(),
-          originalId: conversation.id,
-          folderId: getFolderIdByPath(publishRequest.path, publishedFolders),
-          publishedWithMe: true,
-          name: publishRequest.name,
-          shareUniqueId: publishRequest.shareUniqueId,
-          publishVersion: publishRequest.version,
-        }));
+    switchMap(
+      ({ publishRequest, conversations, publishedAndTemporaryFolders }) => {
+        const sharedConversations = conversations
+          .filter((conversation) => conversation.id === publishRequest.id)
+          .map(({ folderId: _, ...conversation }) => ({
+            ...conversation,
+            ...resetShareEntity,
+            id: uuidv4(),
+            originalId: conversation.id,
+            folderId: getFolderIdByPath(
+              publishRequest.path,
+              publishedAndTemporaryFolders,
+            ),
+            publishedWithMe: true,
+            name: publishRequest.name,
+            shareUniqueId: publishRequest.shareUniqueId,
+            publishVersion: publishRequest.version,
+          }));
 
-      return concat(
-        of(
-          ConversationsActions.addConversations({
-            conversations: sharedConversations,
-          }),
-        ),
-      );
-    }),
+        const temporaryFolders = getTemporaryFoldersToPublish(
+          publishedAndTemporaryFolders,
+          sharedConversations[sharedConversations.length - 1].folderId,
```

Comment:
Are you sure that correct folder will always be on last place i.e. the order is correct?

Reply:
In this case the sequence in the array starts with children and ends with the root folder, but to be sure that it will always be valid, it is really better to make a separate method to search root, thanks!